src/odemis/util/peak.py

```diff
@@ -254,12 +285,27 @@ def _DoFit(self, future, spectrum, wavelength, type='gaussian'):
                     continue
 
                 fit_list = []
+                lower_bounds = []
+                upper_bounds = []
                 for (pos, amplitude) in peaks:
-                    fit_list.append(pos)
-                    fit_list.append(width)
-                    fit_list.append(amplitude)
-                # Initialize offset to 0
-                fit_list.append(0)
+                    if type in {'gaussian_energy', 'lorentzian_energy'}:
+                        fit_list.extend(peak_to_energy(pos, width, amplitude))
+                        # lower & upper bounds for center position, width, amplitude in energy domain
+                        lower_bounds.extend([0, 0, 0])
+                        upper_bounds.extend([100, 10, numpy.inf])
```

Comment:
These constants are not explained enough. You could mention that 100 nm ~= 12 and 1000nm ~= 0.12, so they are very large bounds.
Alternatively, you could use the same formula as for the space domain (but add a `WL_TO_ENERGY / `)
Another way could be to move the space domain formula up, and to call `peak_to_energy()` for the lower and upper bounds too.